src/databricks/labs/ucx/mixins/fixtures.py

```diff
@@ -638,6 +639,33 @@ def _scim_values(ids: list[str]) -> list[iam.ComplexValue]:
     return [iam.ComplexValue(value=x) for x in ids]
 
 
+def wait_group_provisioned(interface: AccountGroupsAPI | GroupsAPI, *groups: Group) -> None:
```

Comment:
encode it into `_make_group`, make waiting by default with opt-out

Reply:
@nfx: I think this PR already does this?

This function is exposed for tests (and fixtures such as `make_ucx_group()`) that create multiple groups so that they can create them all and then wait on them all at once instead of having to wait (~60s) for each one before creating the next group and so forth.

I've added a docstring to emphasise that tests should only use this when creating multiple groups.

Reply:
This method is no longer public, and the fixture always (unconditionally) invokes it after each group is created; callers can no longer defer the waiting.